### PR TITLE
Adds Serializer that converts JSON to Buffer

### PR DESCRIPTION
**Pull request by SentryMan**
When I want to use composite metadata, I have to use the buffer encoders on my transport, but the standard JSON Serializer will then fail because the data objects I send aren't buffers. This serializer is the workaround I came up with. 

Also I added a composite metadata example to the docs

### CODE REVIEW

File: docs/01-client-configuration.md [hunk 1]
```diff
@@ -98,6 +98,58 @@ client.connect().subscribe({
 });
 ```
 
+## Composite Metadata
+To use Composite Metadata, you will need to configure your Rsocket Client as normal, but for the metadata Mime-Type put composite, and ensure your Transport uses `BufferEncoders`
```
Comment (yschimke) [inline docs/01-client-configuration.md:102]:
```suggestion
To use Composite Metadata, you will need to configure your RSocket Client as normal, but for the metadata Mime-Type put composite, and ensure your Transport uses `BufferEncoders`
```

**Comment by SentryMan** [comment]
I'll be honest, I don't know how this DCO thing works

File: packages/rsocket-core/src/RSocketSerialization.js [hunk 2]
```diff
@@ -60,6 +60,63 @@ export const JsonSerializers = {
...
+      str = buffer.toString("utf8");
+    }
+
+    if (str.includes("application/json")) {
+      const objArray = [];
+      const jsonStrArr = str.split("application/json");
+      for (let i = 1; i < jsonStrArr.length; i++) {
+        const unformattedJsonstr = jsonStrArr[i];
+
+        const jsonstr = unformattedJsonstr.substring(
```
Comment (yschimke) [inline packages/rsocket-core/src/RSocketSerialization.js:103]: This deserves a comment or an example.  Not clear why this is needed?

File: packages/rsocket-core/src/RSocketSerialization.js [hunk 3]
```diff
@@ -60,6 +60,63 @@ export const JsonSerializers = {
...
+      str = metadata;
+    } else if (Buffer.isBuffer(metadata)) {
+      const buffer: Buffer = metadata;
+      str = buffer.toString("utf8");
+    } else {
+      const buffer: Buffer = Buffer.from(metadata);
+      str = buffer.toString("utf8");
+    }
+
+    if (str.includes("application/json")) {
```
Comment (yschimke) [inline packages/rsocket-core/src/RSocketSerialization.js:97]: should this be at minimum startsWith? or an equals check? But could you give examples here, not sure being flexible here is a big win.

**Comment by yschimke** [comment]
> I'll be honest, I don't know how this DCO thing works

Think the link explains it "https://github.com/rsocket/rsocket-js/pull/124/checks?check_run_id=2433501124"

**Comment by SentryMan** [comment]
I'm doing the steps in that link but it doesn't resolve anything

**Comment by SentryMan** [comment]
1.0.0 looks like it'll remove serializers

**Pull request closed without merging**